### ISSUE TITLE
[FR] Adding an 'every' parameter to the ModelCheckpoint callback

### ISSUE THREAD

**Issue by alito** [fchollet/keras#4589]
I think adding an 'every' parameter to the ModelCheckpoint callback would be useful, for it to save every N epochs (defaulting to 1 of course).

ie change the constructor to:
```
    def __init__(self, filepath, monitor='val_loss', verbose=0,
                 save_best_only=False, save_weights_only=False,
                 mode='auto', every=1):
```
Sometimes epochs are very short and the loss function is noisy so relying on loss function changes to decide when to save isn't great. 

The code for the patch is trivial. I can open a PR if others think this would be useful

**Comment by DingKe** [comment]
+1. Maybe a more descriptive name (like period, interval)  will be better?

**Comment by fchollet** [comment]
Should be called `frequency` or something to that effect. Why not.

**Comment by alito** [comment]
All of the given suggestions seem like an improvement, although frequency tends to be a measurement of events per time instead of time per event. @fchollet was frequency the final verdict, or would you take period as well?
